includes/class-wc-payments-incentives-service.php

```diff
@@ -77,6 +78,23 @@ public function allowed_promo_notes( $promo_notes = [] ): array {
 		return $promo_notes;
 	}
 
+	/**
+	 * Adds the WooPayments incentive badge to the onboarding task.
+	 *
+	 * @param string $badge Current badge.
+	 *
+	 * @return string
+	 */
+	public function onboarding_task_badge( string $badge ): string {
+		$incentive = $this->get_cached_connect_incentive();
+		// Return early if there is no eligible incentive.
```

Comment:
I would add that there can be wcpay incentive set in WC Core, so we'll just return another possible badge, if connect incentive is not available